File: Microsoft/microsoft/5/5.py
freqMap = {};
setMap = {};
maxfreq = 0;
def push(x):
    global maxfreq;
    if x not in freqMap:
        freqMap[x] = 0
    freq = freqMap[x] + 1;
    freqMap[x] = freq
    if (freq > maxfreq):
        maxfreq = freq
    if freq not in setMap:
        setMap[freq] = []
    setMap[freq].append(x);

def pop():
    global maxfreq
    top = setMap[maxfreq][-1];
    setMap[maxfreq].pop();
    freqMap[top] -= 1;
    if (len(setMap[maxfreq]) == 0):
        maxfreq -= 1;
    return top;

if __name__ == "__main__":
    # Push elements to the stack
    n=int(input())
    for i in range(0,n):
        x=int(input())
        push(x);
    # Pop elements
    print(pop())

File: Microsoft/5/5.cpp
#include <bits/stdc++.h>
using namespace std;
map<int, int> freqMap;
map<int, stack<int>> setMap;
int maxfreq = 0;
void push(int x) {
  int freq = freqMap[x] + 1;
  freqMap[x] = freq;
  if (freq > maxfreq)
    maxfreq = freq;
  setMap[freq].push(x);
}
int pop() {
  if (setMap[maxfreq].size() == 0) {
    return 0;
  }
  int top = setMap[maxfreq].top();
  setMap[maxfreq].pop();
  freqMap[top]--;
  if (setMap[maxfreq].size() == 0)
    maxfreq--;
  return top;
}
int main() {
  int n;
  cin >> n;
  for (int i = 0; i < n; i++) {
    int x;
    cin >> x;
    push(x);
  }
  while (setMap[maxfreq].size() != 0) {
    cout << (pop()) << "\n";
  }
  return 0;
}
